### PR TITLE
Fix parsing bug when error body contains no spaces

### PR DESCRIPTION
**Pull request by charsyam**
This is only copy for https://github.com/twitter/twemproxy/pull/406 (by @tom-dalton-fanduel )
for testing. I just PR again.

### CODE REVIEW

**Comment by charsyam** [comment]
@manjuraj This passed testcases. How about recommiting it?

**Comment by charsyam** [comment]
@manjuraj ping

**Comment by tom-dalton-fanduel** [comment]
+1 from me too!

**Comment by charsyam** [comment]
@manjuraj ping again :)

File: src/proto/nc_redis.c [hunk 1]
```diff
@@ -1920,6 +1920,11 @@ redis_parse_rsp(struct msg *r)
 
                     break;
                 }
+
+                if (ch == CR) {
```
Comment (TysonAndre) [inline src/proto/nc_redis.c:1924]: An alternative from 3e55462c0e5d9d95a88e099a777b7b3ebd373065 in a fork

```
-                state = SW_RUNTO_CRLF;
+                if (ch == '\r') {
+                    state = SW_ALMOST_DONE;
+                } else {
+                    // Read remaining characters until '\r'
+                    state = SW_RUNTO_CRLF;
+                }
             }
```

Both approaches work.

**Comment by TysonAndre** [comment]
The fix was moved to https://github.com/twitter/twemproxy/pull/611 along with a C unit test and updating it to support the newer python/library versions for the integration test, avoiding going backwards in a state machine out of personal preference.

I'd found the same bug before seeing this PR and fixed it in a slightly different way, though both work

**Pull request closed without merging**